tethysext/atcore/services/resource_spatial_manager.py

```diff
@@ -0,0 +1,103 @@
+"""
+********************************************************************************
+* Name: model_db_spatial_manager
+* Author: nswain
+* Created On: July 06, 2018
+* Updated on: December 19, 2018
+* Copyright: (c) Aquaveo 2018
+********************************************************************************
+"""
+import os
+from jinja2 import Template
+
+from abc import abstractmethod
+
+from tethysext.atcore.services.exceptions import UnitsNotFound, UnknownUnits
+from tethysext.atcore.services.base_spatial_manager import BaseSpatialManager
+from tethysext.atcore.models.app_users import Resource
+
+
+class ResourceSpatialManager(BaseSpatialManager):
+    """
+    Class for Spatial Managers using a postgres database.
+    """
+
+    SQL_PATH = os.path.join(os.path.dirname(os.path.dirname(__file__)), 'resources', 'sql_templates')
+    SLD_PATH = os.path.join(os.path.dirname(os.path.dirname(__file__)), 'resources', 'sld_templates')
+
+    # Vector Layer Types
+    VL_EXTENT_VIEW = 'resource_extent_layer_view'
+    VL_EXTENT_TEMPLATE = 'resource_extent_layer_style'
+
+    def get_extent_for_project(self, datastore_name, resource_id):
+        """
+        Get the extent. This will return the list of the extent in EPSG 4326.
+        The query in resource_extent_layer_view transforms all features to 4326.
+
+        Args:
+            datastore_name(str): name of the app database, for example: app_primary_db.
+            resource_id(str): id of the extent record.
+        """
+        # feature name
+        feature_name = f'app_users_resources_{resource_id}'
```

Comment:
`f'app_users_resources_{resource_id}'` is used in a few places. Add this function to the class and use it to get the name of the extent layer:

```python
def get_extent_layer_name(self, resource_id):
    """
    Get name of the extent layer for a given resource.

    Args:
        resource_id(str): id of the Resource.

    Returns:
        str: name of the extent layer.

    """
    return f'app_users_resources_extent_{resource_id}'
```

This will ensure we don't end up adding a bug if we change this in the future. Also, not the addition of "extent" to the layer name.

Reply:
Use this function everywhere we are hardcoding: `f'app_users_resources_{resource_id}'`

Reply:
Please change 'app_users_resources_' to 'app_users_resources_extent_'. This will future proof the layer name if we end up having other layers from the app_users_resources table.